### ISSUE TITLE
Is  this better?

### ISSUE THREAD

**Issue by yuedajiong** [liren2515/DrapeNet#25]
The  range of dgcnn output is very large, useing tanh() to limit the range smaller, is this better?  e.g   tanh(old_output) * C  (to C, for example1.0~10.0)

**Comment by liren2515** [comment]
Hello,

You should use [this function](https://github.com/liren2515/DrapeNet/blob/5fdc6a97e46f90af3c04fff0acbd321fecca8c96/encdec/export_meshes.py#L81) to change the output of dgcnn for meshing.

**Comment by yuedajiong** [comment]
Thanks @liren2515 

A related question:  if I want to use another generation network to generation 'latent' from images, instead of pointcloud, does the latent using this output P_2 (P_2 = (1 - torch.sigmoid(P_1)) * udf_max_dist)) better than P_1?    (My models have 2 network, one is VAE which is from PC to latent to UDF/Mesh, another is GEN which is from images to latent to UDF/Mesh.   network-1 is using to help training latent-udf/mesh decoder).

Thanks.


with torch.no_grad():
    latent_codes = encoder(pcds)

for i in progress_bar(range(bs), "Meshing"):
    lat = latent_codes[i].unsqueeze(0)

def udf_func(c: Tensor) -> Tensor:
    c = coords_encoder.encode(c.unsqueeze(0))      
    p = decoder(c, lat).squeeze(0)                             ### Here-1:   abs(lat) is very large, maybe throunds ...   
    p = torch.sigmoid(p)                                           ### Here-2:  yes, range of p is limited.
    p = (1 - p) * udf_max_dist
    return p

**Comment by liren2515** [comment]
> Thanks [@liren2515](https://github.com/liren2515)
> 
> A related question: if I want to use another generation network to generation 'latent' from images, instead of pointcloud, does the latent using this output P_2 (P_2 = (1 - torch.sigmoid(P_1)) * udf_max_dist)) better than P_1? (My models have 2 network, one is VAE which is from PC to latent to UDF/Mesh, another is GEN which is from images to latent to UDF/Mesh. network-1 is using to help training latent-udf/mesh decoder).
> 
> Thanks.
> 
> with torch.no_grad(): latent_codes = encoder(pcds)
> 
> for i in progress_bar(range(bs), "Meshing"): lat = latent_codes[i].unsqueeze(0)
> 
> def udf_func(c: Tensor) -> Tensor: c = coords_encoder.encode(c.unsqueeze(0)) p = decoder(c, lat).squeeze(0) ### Here-1: abs(lat) is very large, maybe throunds ... p = torch.sigmoid(p) ### Here-2: yes, range of p is limited. p = (1 - p) * udf_max_dist return p

I think it won't have too much difference in terms of accuracy. Using sigmoid helps our model converge faster.

**Comment by yuedajiong** [comment]
Thanks  @liren2515
